### DISCUSSION TITLE
Routing to subdomain not working

### DISCUSSION THREAD

**Discussion by schvila** [dotnet/yarp#2063] [Q&A] [answered]
I want to configure a route to the subdomain, in code, I loaded the certificate and configured, listenOptions.UseHttps:
```c#
options.Listen(IPAddress.Any, 443, listenOptions =>
{
    listenOptions.UseHttps(new HttpsConnectionAdapterOptions()
    {
        ServerCertificate = x509cerificate;
    });
});
```

appsettings.json:
```json
{
    "Logging": {
      "LogLevel": {
        "Default": "Information",
        "Microsoft": "Information",
        "Yarp": "Information",
        "Microsoft.Hosting.Lifetime": "Information",
        "Microsoft.ReverseProxy": "Information"
      }
    },
    "Urls": "https://*.**mydomain.com**",
    "AllowedHosts": "*",
    "ReverseProxy": {
      "portalRoute": {
        "ClusterId": "clusterPortal",
        "Match": {
          "Hosts": [
            "portal.mydomain.com"
          ],
          "Path": "{**catch-all}"
        },
        "transforms": [
          {
            "pathPattern": "{**catch-all}"
          }
        ]
      }
      ,
      "Clusters": {
        "clusterPortal": {
          "Destinations": {
            "destination1": {
              "Address": "https://company.portal.eu:44037"
            }
          }
        }
      }
    }
  }
```

 Attempt to access subdomain using https:// portal.mydomain.com/Check , not routed and not logged.
Testing on customer's azure windows virtual machine. Before YARP we used URL Rewrite in IIS.
Relevant log content:
Overriding address(es) 'https://*.mydomain.com'.
Now listening on: https://0.0.0.0:443

**Comment by samsp-msft** [comment]
Q: It may be syntactic sugar that I am not aware of, but why is URLs specified as "https://*.**mydomain.com**" with the extra * around the domain name?
You should get a log entry from the host for each connection and URL request. That fact that you don't get log entries is concerning. Are you sure that the client is connecting to the server? If testing locally, how have you tweaked the DNS so that Portal.mydomain.com points to the instance.

**Reply by schvila** [reply]
Extra * around the domain name appeared after I used bold formatting. Domain and subdomain names are not real I used them just for problem description. We are in process of switching off IIS on azure virtual machine and using Kestrel services and YARP in front of that.
Tests in aour local virtual dev environmen using portal.company.com routing to Host service.company.com worked properly. Similar configuration in azure is not working may by due to using external IPs for domains, but that works on IIS Url rewriting properly. Right now I am lost as there is nothing in yarp log. Just for test I added Host 'localhost' and send request to https://localhost which appeared in log but requests to https://service.company.com/user is not logged.

**Comment by samsp-msft** [comment]
Please try increasing the log level for "Microsoft" so you can get detailed logs from the host and routing to ensure that requests are coming in. See https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-7.0#configure-logging and https://microsoft.github.io/reverse-proxy/articles/diagnosing-yarp-issues.html

**Comment by Tratcher** [comment]
> Overriding address(es) 'https://*.mydomain.com'.
Now listening on: [https://0.0.0.0:443](https://0.0.0.0/)

> "Urls": "https://*.**mydomain.com**",

Urls is having no effect in this case, Listen is overriding it. That's ok, it wouldn't have done anything differently, kestrel doesn't check the host name. It only has special handling for IPs and localhost.

> Attempt to access subdomain using https:// portal.mydomain.com/Check , not routed and not logged.

What does happen? What response does the client get?

**Comment by schvila** [comment, marked as answer]
Thank you for your help, issue solved - configuration of the azure windows virtual machine netwok. YARProutse properly to subdomins.

**Comment by schvila** [comment]
YARP is ok.
